src/blocking/spi.rs

```diff
@@ -108,3 +108,134 @@ pub mod write_iter {
         }
     }
 }
+
+/// Provides SpiWithCS wrapper around an spi::* and OutputPin impl
+pub mod spi_with_cs {
+
+    use core::marker::PhantomData;
+    use core::fmt::Debug;
+
+    use crate::digital::OutputPin;
+    use super::*;
+
+    /// SpiWithCS wraps an blocking::spi* implementation with Chip Select (CS)
+    /// pin management.
+    /// For sharing SPI between peripherals, see [shared-bus]()
+    pub struct SpiWithCs<Spi, SpiError, Pin, PinError> {
+        spi: Spi,
+        cs: Pin,
+
+        _spi_err: PhantomData<SpiError>,
+        _pin_err: PhantomData<PinError>,
+    }
+
+    /// SpiWithCsErr provies an error numberation over generic Spi and Pin variants
+    #[derive(Clone, Debug, PartialEq)]
+    pub enum SpiWithCsErr<SpiError, PinError> {
```

Comment:
Some bikeshedding: How about `SpiWithCsError`?

Reply:
is `Err` unclear given it's already part of rust nomenclature?

Reply:
It is clear, but all other error types we have here are called `Error` in full which is just 2 more letters. Anyway it is not very important.

Reply:
fair and, renamed!